operator/cmd/mesh/testdata/manifest-generate/input/pilot_override_kubernetes.yaml

```diff
@@ -9,7 +9,6 @@ spec:
   components:
     pilot:
       enabled: true
-      namespace: istio-control
```

Comment:
does it means that the k8s overlay for only the resources in default namespace or cluster scoped ones?

Reply:
my understanding is that resources in non-default namespace still can be patched, as the `defaultNamespace` in these change set is still retrieved from component namespace, right?

Reply:
that's right, the component namespace is used so the only thing you cannot patch is something in a namespace that's not mentioned in IOP.

Reply:
then why we change the ns from non-default here? I thought we want to make sure non-default one also works? or is it covered by other test

Reply:
it's because having istiod in its own namespace like this is not supported and may be confusing to users. but i agree it's better to have the test for the different ns so i've put it back.